src/main/java/com/github/lordrex34/config/supplier/DefaultConfigSupplier.java

```
@@ -0,0 +1,78 @@
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.lordrex34.config.ConfigManager;
+import com.github.lordrex34.config.annotation.ConfigField;
+import com.github.lordrex34.config.converter.IConfigConverter;
+import com.github.lordrex34.config.lang.FieldParser.FieldParserException;
+import com.github.lordrex34.config.util.PropertiesParser;
+
+/**
+ * This is the configuration value supplier used by {@link ConfigField} annotation by default.
+ * @author lord_rex
+ */
+public class DefaultConfigSupplier implements IConfigValueSupplier<Object>
+{
+	private static final Logger LOGGER = LoggerFactory.getLogger(DefaultConfigSupplier.class);
+	
+	/** To avoid creating the same converter thousand times. */
+	private static final Map<String, IConfigConverter> CONVERTERS = new HashMap<>();
+	
+	@Override
+	public Object supply(Field field, ConfigField configField, PropertiesParser properties, PropertiesParser overridenProperties) throws InstantiationException, IllegalAccessException
+	{
+		final String propertyKey = configField.name();
+		final String propertyValue = configField.value();
+		
+		final String configProperty = ConfigManager.getProperty(properties, overridenProperties, propertyKey, propertyValue);
+		
+		final Class<? extends IConfigConverter> converterClass = configField.converter();
+		final String converterClassName = converterClass.getName();
+		IConfigConverter converter = CONVERTERS.get(converterClassName);
+		if (converter == null)
+		{
+			converter = converterClass.newInstance();
+			CONVERTERS.put(converterClassName, converter);
+		}
+		
+		Object value;
+		try
+		{
+			value = converter.convertFromString(field, field.getType(), configProperty);
+		}
+		catch (FieldParserException e)
+		{
+			value = converter.convertFromString(field, field.getType(), propertyValue);
+			LOGGER.warn("Property '{}' has incorrect syntax! Using default value instead: {}", propertyKey, propertyValue);
```

(I would fail here... or at least let the end-use decide)

---

I agree. I would actually throw this error instead of catching it. Let the user handle it from his side.

---

Then it would broke the feature requested by @UnAfraid that if user fails to specify a config, then a default value is supposed to be used.

Remember the issues we had with "outdated configs" whatever?

---

given that issue it's clear that it should fail the shit out of the server... ffs... this is the worst idea like this... I create a typo and instead of the intended value it won't just fail to launch the server, but it will start with the default value? NOPE.

---

Done.